src/briefcase/commands/dev.py

```diff
@@ -74,23 +77,33 @@ def add_options(self, parser):
             help="Run the app in test mode",
         )
 
-    def install_dev_requirements(self, app: AppConfig, **options):
+    def install_dev_requirements(
+        self, app: AppConfig, venv_path: Path | None = None, **options
+    ):
         """Install the requirements for the app dev.
 
         This will always include test requirements, if specified.
 
         :param app: The config object for the app
         """
+
         requires = app.requires if app.requires else []
         if app.test_requires:
             requires.extend(app.test_requires)
 
         if requires:
             with self.console.wait_bar("Installing dev requirements..."):
                 try:
+                    py_exe = (
+                        os.fspath(venv_python(venv_path))
+                        if venv_path
+                        else sys.executable
+                    )
```

Comment:
This is the sort of logic where I was suggesting having an object representation of a "python environment" might make sense. If `venv` is the object returned by a virtual environment context manager, then `venv.executable` is the executable for that environment - and the no-op venv just returns `sys.executable`; `venv.env` is the environment variables needed to enable that venv (or is a method that can be invoked with an existing environment to "virtualise" it.)